library/core/src/mem/maybe_uninit.rs

```diff
@@ -374,6 +374,20 @@ impl<T> MaybeUninit<T> {
     /// assert_eq!(x, (0, false));
     /// ```
     ///
+    /// This can be used in const contexts, such as to indicate the end of plugin registration
+    /// arrays.
+    ///
+    /// ```
+    /// use std::mem::MaybeUninit;
+    ///
+    /// struct PluginInfo { id: u32, action: Option<fn(u32) -> u32> }
+    ///
+    /// static PLUGIN_LIST: [PluginInfo; 2] = [
+    ///     PluginInfo { id: 1, action: Some(|x| x + 5) },
+    ///     unsafe { MaybeUninit::zeroed().assume_init() }
```

Comment:
nit: could we find a better example? This could be `PluginInfo { id: 0, action: None }`, which is both shorter *and* doesn't need `unsafe`.

(And please put a `// SAFETY: …` comment on any `unsafe { … }` used in examples, to encourage good behaviour and explain why it's not UB.)

Reply:
Sure, any suggestions? I was struggling to come up with something better, null-terminated arrays for plugins are just the use case I always come across

Reply:
The entire problem with a good example here is that you use zeroed when the full thing would be a pain to write or even look at. So usually that means that you're talking about some type with 50 integer fields or something. But regardless of details it's "something so big it doesn't fit in a 2 line example".

Reply:
Are we able to use dev dependencies? Is there a `libc` use case to feature here?

Reply:
SafeZeroed would actually have to be an unsafe trait in this scenario.

Honestly I don't think we need example code on this particular function/method. It makes the all-zeroed value, that's already extremely simple I think.

Reply:
I really just wanted a sanity check test for the functionality, but agree it’s redundant. I’ll remove the example and change it to a unit test.

(Not in front of a computer now but I will do it tomorrow)

Reply:
Nevermind, I was able to do it now

@dtolnay`bot` ready 🙂